airbyte-workers/src/test/java/io/airbyte/workers/internal/EmptyAirbyteSourceTest.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.internal;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.ResetSourceConfiguration;
+
+import io.airbyte.config.State;
+import io.airbyte.config.WorkerSourceConfig;
+import io.airbyte.protocol.models.AirbyteGlobalState;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteMessage.Type;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStreamState;
+import io.airbyte.protocol.models.StreamDescriptor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class EmptyAirbyteSourceTest {
+
+  private EmptyAirbyteSource emptyAirbyteSource;
+  private final AirbyteMessage EMPTY_MESSAGE =
+      new AirbyteMessage().withType(Type.STATE)
+          .withState(new AirbyteStateMessage().withStateType(AirbyteStateType.LEGACY).withData(Jsons.emptyObject()));
+
+  @BeforeEach
+  public void init() {
+    emptyAirbyteSource = new EmptyAirbyteSource();
+  }
+
+  @Test
+  public void testLegacy() throws Exception {
+    emptyAirbyteSource.start(new WorkerSourceConfig(), null);
+
+    legacyStateResult();
+  }
+
+  @Test
+  public void testEmptyListOfStreams() throws Exception {
+    ResetSourceConfiguration resetSourceConfiguration = new ResetSourceConfiguration()
+        .withStreamsToReset(new ArrayList<>());
+    WorkerSourceConfig workerSourceConfig = new WorkerSourceConfig()
+        .withSourceConnectionConfiguration(Jsons.jsonNode(resetSourceConfiguration));
+
+    emptyAirbyteSource.start(workerSourceConfig, null);
+
+    legacyStateResult();
+  }
+
+  @Test
+  public void nonStartedSource() {
+    Throwable thrown = Assertions.catchThrowable(() -> emptyAirbyteSource.attemptRead());
+    Assertions.assertThat(thrown)
+        .isInstanceOf(IllegalStateException.class);
+  }
+
+  @Test
+  public void testGlobal() throws Exception {
+    List<StreamDescriptor> streamDescriptors = Lists.newArrayList(
+        new StreamDescriptor().withName("a"),
+        new StreamDescriptor().withName("b"),
+        new StreamDescriptor().withName("c")
+    );
+
+    List<io.airbyte.config.StreamDescriptor> streamToReset = Lists.newArrayList(
```

Comment:
is this stream the same as the previous one? do we just need a helper to change the types?